dangerzone/isolation_provider/base.py

```diff
@@ -93,10 +93,6 @@ def __init__(self) -> None:
         else:
             self.proc_stderr = subprocess.DEVNULL
 
-    @staticmethod
```

Comment:
> At the moment, the Isolation Providers do not expose a clear external interface, and as a result we end up consuming methods that are part of the interface and methods which aren't.

The way I see, any method defined in the abstract `base.IsolationProvider` is part of the public interface, and methods defined in the concrete classes are implementation-specific. Is there any other place where we violate that? I'd argue that we _can_ make `is_runtime_available()` a part of the public interface, but we don't gain much from that.

> Rather than doing isinstance calls, add the missing concepts in our interface.

In our code, we have two `isinstance()` calls in order to differentiate between adding a waiting widget for containers, or not. We could turn these `isinstance()` calls to an `IsolationProvider.requires_wait()` method, if it looks good to you.

Also, my PR adds a third one, but that's just sugar for the `mypy`, which doesn't get that this code path is only reachable by `Container`/`Dummy` isolation providers, who have the `is_runtime_available()` method. I could add a `# type: ignore ...` comment to placate `mypy`, but I wanted to make things more explicit.

Reply:
We could refactor this in the future, maybe in a pair programming session together so we agree upon something! Closing this for now, as it's not blocking.

Reply:
Damn, just did it :grimacing:. Anyways, I'll push it soon, and you can let me know if it works for you.